src/libs/ajax/workflows-app/Cbas.test.js

```diff
@@ -0,0 +1,337 @@
+import 'setimmediate';
+
+import { MatchersV3, PactV3, SpecificationVersion } from '@pact-foundation/pact';
+import path from 'path';
+import { fetchFromProxy, fetchOk, jsonBody } from 'src/libs/ajax/ajax-common';
+import { Cbas } from 'src/libs/ajax/workflows-app/Cbas';
+import {
+  runSetInputDef,
+  runSetInputDefWithSourceNone,
+  runSetInputDefWithStruct,
+  runSetOutputDef,
+} from 'src/libs/ajax/workflows-app/CbasMockResponses';
+
+jest.mock('src/libs/ajax/ajax-common');
+
+jest.mock('src/libs/auth', () => {
+  return {
+    reloadAuthToken: jest.fn(),
+    signOutAfterSessionTimeout: jest.fn(),
+  };
+});
+
+jest.mock('src/libs/state', () => {
+  return {
+    ...jest.requireActual('src/libs/state'),
+    getUser: jest.fn(() => ({ token: 'testtoken' })),
+  };
+});
+
+const { timestamp, string, regex, boolean, integer, fromProviderState } = MatchersV3;
+
+const UUID_REGEX = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}';
+const RUN_STATE_REGEX = 'UNKNOWN|QUEUED|INITIALIZING|RUNNING|PAUSED|COMPLETE|EXECUTOR_ERROR|SYSTEM_ERROR|CANCELED|CANCELING';
+const RUNSET_STATE_REGEX = 'UNKNOWN|QUEUED|RUNNING|COMPLETE|ERROR|CANCELED|CANCELING';
+
+const cbasPact = new PactV3({
+  consumer: 'terra-ui',
+  provider: 'cbas',
+  log: path.resolve(process.cwd(), 'logs', 'pact.log'),
+  logLevel: 'warn',
+  dir: path.resolve(process.cwd(), 'pacts'),
+  spec: SpecificationVersion.SPECIFICATION_VERSION_V3,
+});
+
+describe('Cbas tests', () => {
+  it('should GET run_sets with method ID 00000000-0000-0000-0000-000000000009', async () => {
+    const expectedResponse = {
+      fully_updated: boolean(true),
+      run_sets: [
+        {
+          run_set_id: regex(UUID_REGEX, '10000000-0000-0000-0000-000000000009'),
+          method_id: regex(UUID_REGEX, '00000000-0000-0000-0000-000000000009'),
+          method_version_id: regex(UUID_REGEX, '90000000-0000-0000-0000-000000000009'),
+          is_template: boolean(true),
+          run_set_name: string('struct_workflow_test template run set'),
+          run_set_description: string('struct_workflow_test template submission'),
+          state: regex(RUNSET_STATE_REGEX, 'COMPLETE'),
+          record_type: string('sample'),
+          submission_timestamp: timestamp("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", '2023-03-28T13:05:02.690+00:00'),
```

Comment:
When running this locally, verification fails on these two timestamps. Wasn't able to determine exactly why.